### PR TITLE
MTN use new xfail functionality to skip cyanure/lightning and julia on OSX

### PR DESCRIPTION
**Pull request by tomMoral**

### CODE REVIEW

**Comment by tomMoral** [comment]
@tanglef if you have an idea of what could cause the failure, feel free to ping me :)
I know that you have some expertise with debugging `rpy2` ;)

**Comment by tanglef** [comment]
So I looked a little into that. And in the log there's an issue with the dynamic loading (and shared objects).
Following [this issue (stil opened though...)](https://github.com/rpy2/rpy2/issues/771), maybe setting the LD_LIBRARY_PATH like  [that (more or less)](https://github.com/tanglef/benchOpt/blob/fe08d64bbb3a0708210e749c86db15e755bf0d46/benchopt/helpers/r_lang.py#L29) might help to connect things up again

**Review by agramfort — APPROVED** [review]
I let you merge when happy @tomMoral :)

File: test_config.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    if 'julia' in solver_class.name.lower() and sys.platform == 'darwin':
+        pytest.xfail('Julia causes segfault on OSX for now.')
+
+    # Lightning install is broken on python3.9+.
+    # See issue scikit-learn-contrib/lightning#153.
+    if (solver_class.name.lower() == 'lightning'
+            and sys.version_info >= (3, 9)):
+        pytest.xfail('Lightning install is broken on python3.9+.')
+
+    # Lightning install is broken on python3.9+.
```
Comment (tomMoral) [inline test_config.py:23]: bad coment

**Review by tomMoral — COMMENTED** [review]
oops!

**Pull request merged**